scripts/core/ui/components/PlainTextEditor/PlainTextEditor.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import {
+    Editor,
+    EditorState,
+    ContentState,
+    Modifier,
+    SelectionState,
+    CompositeDecorator,
+} from 'draft-js';
+import './styles.scss';
+import {getSpellchecker} from 'core/editor3/components/spellchecker/default-spellcheckers';
+import {
+    getSpellcheckWarningsByBlock,
+    getSpellcheckingDecorator,
+} from 'core/editor3/components/spellchecker/SpellcheckerDecorator';
+import {getDraftSelectionForEntireContent} from 'core/editor3/helpers/getDraftSelectionForEntireContent';
+
+export interface IProps {
+    value: string;
+    onChange: (newValue: string, props: IProps) => void;
+    onChangeData: any;
+    classes: string;
+    spellcheck?: boolean;
+    language?: string;
+    placeholder?: string;
+}
+
+interface IState {
+    editorState: EditorState;
+    hasFocus: boolean;
+}
+
+export class PlainTextEditor extends React.Component<IProps, IState> {
```

Comment:
Does the component work well as a standalone react component not connected to angular?

Reply:
It should, I didn't use it without angular but it doesn't rely on anything angular-related.